File: internal/content/testdata/bench/java/nc-validate-bst.java
class Solution {
    public boolean is_valid_bst(TreeNode root) {
        return valid(root, Long.MIN_VALUE, Long.MAX_VALUE);
    }
    private boolean valid(TreeNode node, long lo, long hi) {
        if (node == null) return true;
        if (!(lo < node.val && node.val < hi)) return false;
        return valid(node.left, lo, node.val) && valid(node.right, node.val, hi);
    }
}

File: internal/content/testdata/bench/go/nc-validate-bst.go
func is_valid_bst(root *TreeNode) bool {
	var valid func(node *TreeNode, lo, hi int) bool
	valid = func(node *TreeNode, lo, hi int) bool {
		if node == nil {
			return true
		}
		if node.Val <= lo || node.Val >= hi {
			return false
		}
		return valid(node.Left, lo, node.Val) && valid(node.Right, node.Val, hi)
	}
	return valid(root, -1<<62, 1<<62)
}
